packages/lesswrong/server/autocompleteEndpoint.ts

```diff
@@ -0,0 +1,428 @@
+  while (currentComment.parentCommentId && currentComment.parentCommentId !== currentComment._id) {
+    currentComment = await Comments.findOne({ _id: currentComment.parentCommentId });
+    if (currentComment) {
+      parentComments.push(currentComment);
+    } else {
+      break;
+    }
+  }
+
+  return parentComments.reverse();
+}
+
+const getPostBodyFormatted = (post: DbPost, revisionsMap: Map<string, DbRevision>, authorsMap: Map<string, DbUser>) => {
+  const markdownBody = turndownService.turndown(revisionsMap.get(post._id)?.html ?? "<not available/>");
+  const author = authorsMap.get(post.userId)?.displayName;
+  return `${post.title}
+by ${author}
+${post.baseScore}
+${markdownBody}`.trim();
+}
+
+const getCommentBodyFormatted = (comment: DbComment, revisionsMap: Map<string, DbRevision>, authorsMap: Map<string, DbUser>, parentPost: DbPost) => {
+  const markdownBody = turndownService.turndown(revisionsMap.get(comment._id)?.html ?? comment.contents?.html ?? "<not available/>");
+  const dateString = formatRelative(new Date(comment.createdAt), new Date(), false)
+  const author = authorsMap.get(comment.userId)?.displayName;
+  return `Comment on ${parentPost.title}
+${author} ${dateString} ${comment.baseScore} ${comment.extendedScore?.agreement}
+${markdownBody}`.trim();
+}
+
+const getPostReplyMessageFormatted = (post: DbPost, revisionsMap: Map<string, DbRevision>, authorsMap: Map<string, DbUser>, currentUser: DbUser, prefix: string) => {
+  return `${getPostBodyFormatted(post, revisionsMap, authorsMap)}
+---
+Comment on ${post.title}
+${currentUser.displayName} 1h ${Math.floor(20 + (Math.random() * 75))} ${Math.floor((Math.random() - 0.5) * 100)}
+${prefix}`.trim();
+}
+
+const getCommentReplyMessageFormatted = (comment: DbComment, parentComments: DbComment[], parentPost: DbPost, revisionsMap: Map<string, DbRevision>, authorsMap: Map<string, DbUser>, prefix: string, currentUser: DbUser) => {
+  const parentComment = parentComments[parentComments.length - 1];
+  return `${getPostBodyFormatted(parentPost, revisionsMap, authorsMap)}
+---
+${parentComments.map((comment) => getCommentBodyFormatted(comment, revisionsMap, authorsMap, parentPost)).join("\n---\n")}
+---
+${getCommentBodyFormatted(comment, revisionsMap, authorsMap, parentPost)}
+---
+Comment on ${parentPost.title}
+${currentUser.displayName} 1h ${Math.floor(20 + (Math.random() * 75))} ${Math.floor((Math.random() - 0.5) * 100)}
+${prefix}`.trim();
+}
+
+async function constructMessageHistory(
+  prefix: string,
+  commentIds: string[],
+  postIds: string[],
+  currentUser: any,
+  replyingCommentId?: string,
+  postId?: string
+): Promise<PromptCachingBetaMessageParam[]> {
+  const messages: PromptCachingBetaMessageParam[] = [];
+
+  // Make the fetches parallel to save time
+  const [revisions, posts, comments] = await Promise.all([
+    Revisions.find({ documentId: { $in: [...postIds, ...commentIds] }, fieldName: "contents", version: "1.0.0" }).fetch(),
```

Comment:
why is this `version: "1.0.0"`?

Reply:
Also why are we fetching revisions separately from posts/comments?  You can use `fetchFragment` (or `runQuery`, if you also want to get e.g. the post authors, without needing to fix some bugs in `fetchFragment` that don't handle some resolver fields correctly)